internal/ethapi/api.go

```diff
@@ -1308,25 +1308,20 @@ func newRPCTransaction(tx *types.Transaction, blockHash common.Hash, blockNumber
 		result.BlockNumber = (*hexutil.Big)(new(big.Int).SetUint64(blockNumber))
 		result.TransactionIndex = (*hexutil.Uint64)(&index)
 	}
+
 	switch tx.Type() {
 	case types.AccessListTxType:
 		al := tx.AccessList()
 		result.Accesses = &al
 		result.ChainID = (*hexutil.Big)(tx.ChainId())
+
 	case types.DynamicFeeTxType:
 		al := tx.AccessList()
 		result.Accesses = &al
 		result.ChainID = (*hexutil.Big)(tx.ChainId())
 		result.GasFeeCap = (*hexutil.Big)(tx.GasFeeCap())
 		result.GasTipCap = (*hexutil.Big)(tx.GasTipCap())
-		// if the transaction has been mined, compute the effective gas price
-		if baseFee != nil && blockHash != (common.Hash{}) {
```

Comment:
Note: subtle change here. We previously returned the effective gas price only for already-mined transactions. With this change, we now return it for any block that has a baseFee.

I'm really unsure if this is a problem or not, just made this change because it's more logical to use the `EffectiveGasTip` here.

Reply:
I'm a little confused with this change. I think `GasPrice` should be total price per gas the tx expends towards both the `base_fee` and `tip`? The reason it was the max value before the tx was mined was because it was, at worst, an overestimate. 

This change seems to return a slightly different quantity now, the effective tip paid. So all applications that expect `GasPrice` to be the *total* price per gas expended by transaction will now get an incorrect number.

Reply:
I'm also not sure about this. I think `GasPrice` is a very well defined thing: it's the actual price per unit of gas that the tx paid -- regardless of what the miner got and what was burned. 
For 1559, it can only ever be returned in the context of the block it was executed in, which I don't see any way to work around. However we solve it, it becomes a weird UX, and I don't see how the change in this PR makes the UX any better (but again, I'm not sure -- which means I'm very open to being convinced otherwise if I'm missing something)